spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -456,10 +481,26 @@ def zoom_out(self):
 
     def scale_image(self):
         """Scale the image size."""
-        new_width = int(self.figcanvas.fwidth *
-                        self._scalestep ** self._scalefactor)
-        new_height = int(self.figcanvas.fheight *
-                         self._scalestep ** self._scalefactor)
+        fwidth = self.figcanvas.fwidth
+        fheight = self.figcanvas.fheight
+
+        # Don't auto fit plotting
+        if not self.auto_fit_plotting:
+            new_width = int(fwidth * self._scalestep ** self._scalefactor)
+            new_height = int(fheight * self._scalestep ** self._scalefactor)
+
+        # Auto fit plotting
+        # Scale the image to fit figcanvas size while respect the ratio
+        else:
+            size = self.size()
+            width = size.width() - 15  # How to get the actual figcanvas size?
```

Comment:
@jnsebgosselin ?

Reply:
@jnsebgosselin, I think we need your help to clarify this. Otherwise, @CAM-Gerlach has done a great job reviewing this PR.

Reply:
There's a similar comment in the tests, too, that was added after I reviewed FYI.

Reply:
I think I have made this right, also with the test.